src/lightning/pytorch/loops/evaluation_loop.py

```diff
@@ -114,12 +126,23 @@ def run(self) -> List[_OUT_DICT]:
         previous_dataloader_idx = 0
         while True:
             try:
-                batch, batch_idx, dataloader_idx = next(data_fetcher)
+                if self._is_sequential:
```

Comment:
I think these branches make the code hard to read (again). Shouldn't we try keep this simple? Can't we somehow make it return a standardized way (batch, batch_idx, dataloader_idx) for both cases?

Reply:
We cannot do it without breaking backwards compatibility for those using `dataloader_iter` in their step methods

And the combinedloader/mode iterators would need to know about the batch_progress and the fact that some modes don't require a dataloader_idx

So the reading complexity would still exist, just elsewhere